python/ribasim/ribasim/geometry/edge.py

```diff
@@ -112,13 +123,62 @@ def add(
             index=pd.Index([edge_id], name="edge_id"),
         )
 
+        self._validate_edge(to_node, from_node, edge_type)
+
         self.df = GeoDataFrame[EdgeSchema](pd.concat([self.df, table_to_append]))
         if self.df.duplicated(subset=["from_node_id", "to_node_id"]).any():
             raise ValueError(
                 f"Edges have to be unique, but edge with from_node_id {from_node.node_id} to_node_id {to_node.node_id} already exists."
             )
         self._used_edge_ids.add(edge_id)
 
+    def _validate_edge(self, to_node: NodeData, from_node: NodeData, edge_type: str):
```

Comment:
You don't use edge_type now? Might be good to actually use it, so you can reduce the repetition here, and then you also don't need to do `& (
            edge_type == "flow"` in the final checks.

Reply:
Good point, I am using the edge_type yes. :D

Reply:
You could still further reduce the code here by only using the if/else to determine whether you need flow_edge_neighbor_amount or control_edge_neighbor_amount, and using those in 2 checks insteads of the 4 now.